mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala

```diff
@@ -278,6 +279,7 @@ class BisectingKMeans @Since("2.0.0") (
     val summary = new BisectingKMeansSummary(
       model.transform(dataset), $(predictionCol), $(featuresCol), $(k))
     model.setSummary(Some(summary))
+    instr.logNamedValue("clusterSizes", summary.clusterSizes.mkString(", "))
```

Comment:
This requires parsing the result string if we want to do some analysis. How do we do this in other places? Do we always log in JSON format, e.g.? cc: @WeichenXu123 @jkbradley

Reply:
We log Params in JSON format, but we don't have a precedent for logging Array values.  I'd be OK with ducking the issue by not logging this value, or with logging it as JSON.  If the latter, then we could add a method Instrumentation.logNamedValue taking Array types.

Reply:
sg. @ludatabricks could you update the value to `mkString("[", ",", "]")` for now and leave a TODO to extend `logNamedValue`?